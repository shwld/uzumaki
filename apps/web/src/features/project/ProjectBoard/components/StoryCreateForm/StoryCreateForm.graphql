fragment StoryCreateForm_Item on Story {
  id
  title
  description
  state
  kind
  points
  requesterId
  projectId
  releaseDate
  position
  priority
  createdAt
  updatedAt
  isUnEstimated
  isCompleted
  isDeleted
}

mutation StoryCreateForm_CreateStory($input: CreateStoryInput!) {
  createStory(input: $input) {
    ... on CreateStorySuccessResult {
      result {
        ...StoryCreateForm_Item
      }
    }
    ... on InvalidArgumentsResult {
      issues {
        field
        message
      }
    }
  }
}
