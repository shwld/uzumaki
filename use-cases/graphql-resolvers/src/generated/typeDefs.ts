import gql from 'graphql-tag';
export const typeDefs = gql`
  schema {
    query: Query
    mutation: Mutation
    subscription: Subscription
  }
  type Account implements Node {
    createdAt: DateTime!
    id: ID!
    isDeleted: Boolean!
    name: String!
    projects: ProjectConnection!
    updatedAt: DateTime!
  }
  type AccountConnection implements Connection {
    edges: [AccountEdge]
    pageInfo: PageInfo
  }
  type AccountEdge implements Edge {
    cursor: String
    node: Account
  }
  type Anonymous implements Node {
    id: ID!
  }
  type AnonymousConnection implements Connection {
    edges: [AnonymousEdge]
    pageInfo: PageInfo
  }
  type AnonymousEdge implements Edge {
    cursor: String
    node: Anonymous
  }
  interface Connection {
    edges: [Edge]
    pageInfo: PageInfo
  }
  input CreateAccountInput {
    id: ID!
    name: String!
  }
  union CreateAccountMutationResult =
      CreateAccountSuccessResult
    | InvalidArgumentsResult
    | UnauthorizedResult
  type CreateAccountSuccessResult {
    result: Account!
  }
  input CreateProjectInput {
    accountId: ID!
    description: String
    id: ID!
    initialVelocity: Int!
    name: String!
    privacy: ProjectPrivacy!
  }
  union CreateProjectMutationResult =
      CreateProjectSuccessResult
    | InvalidArgumentsResult
    | UnauthorizedResult
  type CreateProjectSuccessResult {
    result: Project!
  }
  input CreateStoryInput {
    description: String!
    id: ID!
    kind: StoryKind!
    points: Int
    position: StoryPosition!
    priority: Int!
    projectId: ID!
    releaseDate: DateTime
    requesterId: ID!
    state: StoryState!
    title: String!
  }
  union CreateStoryMutationResult =
      CreateStorySuccessResult
    | InvalidArgumentsResult
    | UnauthorizedResult
  type CreateStorySuccessResult {
    result: Story!
  }
  scalar DateTime
  enum DayOfWeek {
    FRIDAY
    MONDAY
    SATURDAY
    SUNDAY
    THURSDAY
    TUESDAY
    WEDNESDAY
  }
  input DestroyStoryInput {
    id: ID!
  }
  union DestroyStoryMutationResult =
      DestroyStorySuccessResult
    | InvalidArgumentsResult
    | UnauthorizedResult
  type DestroyStorySuccessResult {
    result: Story!
  }
  interface Edge {
    cursor: String
    node: Node
  }
  input EstimateStoryInput {
    id: ID!
    points: Int
  }
  union EstimateStoryMutationResult =
      EstimateStorySuccessResult
    | InvalidArgumentsResult
    | UnauthorizedResult
  type EstimateStorySuccessResult {
    result: Story!
  }
  type InvalidArgumentsResult {
    issues: [ValidationIssue!]!
  }
  input InviteProjectMemberInput {
    id: ID!
    projectId: ID!
    role: ProjectMemberRole!
    userEmail: String!
  }
  union InviteProjectMemberMutationResult =
      InvalidArgumentsResult
    | InviteProjectMemberSuccessResult
    | UnauthorizedResult
  type InviteProjectMemberSuccessResult {
    result: ProjectMemberInvitation
  }
  type JoinProjectMemberAlreadyJoinedResult {
    result: ProjectMember!
  }
  input JoinProjectMemberInput {
    confirmationToken: String!
    id: ID!
  }
  union JoinProjectMemberMutationResult =
      InvalidArgumentsResult
    | JoinProjectMemberAlreadyJoinedResult
    | JoinProjectMemberSuccessResult
    | JoinProjectMemberTokenIsAlreadyUsedResult
    | JoinProjectMemberTokenIsExpiredResult
    | UnauthorizedResult
  type JoinProjectMemberSuccessResult {
    result: ProjectMember!
  }
  type JoinProjectMemberTokenIsAlreadyUsedResult {
    result: ProjectMemberInvitation!
  }
  type JoinProjectMemberTokenIsExpiredResult {
    expiredAt: DateTime!
  }
  input MoveStoriesInput {
    projectId: ID!
    stories: [MoveStoriesStoryDestination!]!
  }
  union MoveStoriesMutationResult =
      InvalidArgumentsResult
    | MoveStoriesSuccessResult
    | UnauthorizedResult
  input MoveStoriesStoryDestination {
    id: ID!
    position: StoryPosition!
    priority: Int!
  }
  type MoveStoriesSuccessResult {
    result: [Story!]!
  }
  type Mutation {
    createAccount(input: CreateAccountInput!): CreateAccountMutationResult!
    createProject(input: CreateProjectInput!): CreateProjectMutationResult!
    createStory(input: CreateStoryInput!): CreateStoryMutationResult!
    destroyStory(input: DestroyStoryInput!): DestroyStoryMutationResult!
    estimateStory(input: EstimateStoryInput!): EstimateStoryMutationResult!
    inviteProjectMember(
      input: InviteProjectMemberInput!
    ): InviteProjectMemberMutationResult!
    joinProjectMember(
      input: JoinProjectMemberInput!
    ): JoinProjectMemberMutationResult!
    moveStories(input: MoveStoriesInput!): MoveStoriesMutationResult!
    updateAccount(input: UpdateAccountInput!): UpdateAccountMutationResult!
    updateStory(input: UpdateStoryInput!): UpdateStoryMutationResult!
    updateStoryState(
      input: UpdateStoryStateInput!
    ): UpdateStoryStateMutationResult!
  }
  interface Node {
    id: ID!
  }
  type PageInfo {
    endCursor: String
    hasNextPage: Boolean!
    hasPreviousPage: Boolean
    startCursor: String
  }
  interface PagedConnection {
    nodes: [Node]
    pageInfo: PagedPageInfo
  }
  type PagedPageInfo {
    hasNextPage: Boolean!
    hasPreviousPage: Boolean
    totalPagesCount: Int
  }
  type Project implements Node {
    accountId: ID!
    boardConfig: ProjectBoardConfig!
    boardStatus: ProjectBoardStatus!
    createdAt: DateTime!
    description: String!
    id: ID!
    invitations: ProjectMemberInvitationConnection!
    isDeleted: Boolean!
    members: ProjectMemberConnection!
    name: String!
    privacy: ProjectPrivacy!
    stories(
      after: String
      first: Int
      page: Int
      position: ProjectStoriesSearchPosition
    ): StoryConnection!
    story(id: ID!): Story
    updatedAt: DateTime!
  }
  type ProjectBoardConfig implements Node {
    createdAt: DateTime!
    id: ID!
    initialVelocity: Int!
    iterationLength: Int!
    startIterationOn: DayOfWeek!
    startOn: DateTime
    updatedAt: DateTime!
  }
  type ProjectBoardStatus implements Node {
    createdAt: DateTime!
    id: ID!
    updatedAt: DateTime!
    velocity: Int!
  }
  type ProjectConnection implements Connection {
    edges: [ProjectEdge]
    pageInfo: PageInfo
  }
  type ProjectEdge implements Edge {
    cursor: String
    node: Project
  }
  type ProjectMember implements Node {
    avatarImageUrl: String!
    createdAt: DateTime!
    id: ID!
    isMe: Boolean!
    name: String!
    role: ProjectMemberRole!
    updatedAt: DateTime!
  }
  type ProjectMemberConnection implements Connection {
    edges: [ProjectMemberEdge]
    pageInfo: PageInfo
  }
  type ProjectMemberEdge implements Edge {
    cursor: String
    node: ProjectMember
  }
  type ProjectMemberInvitation implements Node {
    createdAt: DateTime!
    email: String!
    id: ID!
    isJoined: Boolean!
    projectName: String!
    role: ProjectMemberRole!
    updatedAt: DateTime!
  }
  type ProjectMemberInvitationConnection implements Connection {
    edges: [ProjectMemberInvitationEdge]
    pageInfo: PageInfo
  }
  type ProjectMemberInvitationEdge implements Edge {
    cursor: String
    node: ProjectMemberInvitation
  }
  type ProjectMemberInvitationToken implements Node {
    confirmationToken: String!
    createdAt: DateTime!
    expiredAt: DateTime!
    id: ID!
    invitation: ProjectMemberInvitation!
    isExpired: Boolean!
    updatedAt: DateTime!
  }
  type ProjectMemberInvitationTokenConnection implements Connection {
    edges: [ProjectMemberInvitationTokenEdge]
    pageInfo: PageInfo
  }
  type ProjectMemberInvitationTokenEdge implements Edge {
    cursor: String
    node: ProjectMemberInvitationToken
  }
  enum ProjectMemberRole {
    MEMBER
    OWNER
    VIEWER
  }
  enum ProjectPrivacy {
    PRIVATE
    PUBLIC
  }
  enum ProjectStoriesSearchPosition {
    BACKLOG
    DONE
    ICEBOX
  }
  type Query {
    anonymous: Anonymous
    node(id: ID!): Node
    viewer: Viewer
  }
  type Story implements Node {
    createdAt: DateTime!
    description: String!
    id: ID!
    isCompleted: Boolean!
    isDeleted: Boolean!
    isUnEstimated: Boolean!
    kind: StoryKind!
    owners: [User!]!
    points: Int
    position: StoryPosition!
    priority: Int!
    project: Project
    projectId: ID!
    releaseDate: DateTime
    requester: User
    requesterId: ID!
    state: StoryState!
    title: String!
    updatedAt: DateTime!
  }
  type StoryConnection implements Connection {
    edges: [StoryEdge]
    pageInfo: PageInfo
  }
  type StoryEdge implements Edge {
    cursor: String
    node: Story
  }
  enum StoryKind {
    BUG
    CHORE
    FEATURE
    RELEASE
  }
  enum StoryPosition {
    BACKLOG
    CURRENT
    DONE
    ICEBOX
  }
  enum StoryState {
    ACCEPTED
    DELIVERED
    FINISHED
    REJECTED
    STARTED
    UNSTARTED
  }
  type Subscription {
    subscribeStoryUpdate(projectId: ID!): Story
  }
  type UnauthorizedResult {
    errorMessage: String!
  }
  input UpdateAccountInput {
    id: ID!
    name: String!
  }
  union UpdateAccountMutationResult =
      InvalidArgumentsResult
    | UnauthorizedResult
    | UpdateAccountSuccessResult
  type UpdateAccountSuccessResult {
    result: Account!
  }
  input UpdateStoryInput {
    description: String!
    id: ID!
    kind: StoryKind!
    points: Int
    releaseDate: DateTime
    requesterId: ID!
    state: StoryState!
    title: String!
  }
  union UpdateStoryMutationResult =
      InvalidArgumentsResult
    | UnauthorizedResult
    | UpdateStorySuccessResult
  input UpdateStoryStateInput {
    id: ID!
    state: StoryState!
  }
  union UpdateStoryStateMutationResult =
      InvalidArgumentsResult
    | UnauthorizedResult
    | UpdateStoryStateSuccessResult
  type UpdateStoryStateSuccessResult {
    effectedStories: [Story!]!
    result: Story!
  }
  type UpdateStorySuccessResult {
    effectedStories: [Story!]!
    result: Story!
  }
  type User implements Node {
    id: ID!
    name: String!
  }
  type UserConnection implements Connection {
    edges: [UserEdge]
    pageInfo: PageInfo
  }
  type UserEdge implements Edge {
    cursor: String
    node: User
  }
  type ValidationIssue {
    field: String
    message: String
  }
  type Viewer {
    accounts(after: String, first: Int, page: Int): AccountConnection!
    avatarImageUrl: String!
    createdAt: DateTime!
    email: String!
    id: ID!
    invitationToken(confirmationToken: String!): ProjectMemberInvitationToken
    project(id: ID!): Project
    updatedAt: DateTime!
  }
`;
