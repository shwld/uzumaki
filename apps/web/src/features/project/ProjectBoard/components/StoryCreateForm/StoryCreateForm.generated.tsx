import * as Types from '../../../../../graphql/generated/graphql';

import gql from 'graphql-tag';
import * as Urql from 'urql';
export type DateString = string & { __dateStringBrand: any };
export type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;
export type StoryCreateForm_ItemFragment = {
  __typename?: 'Story';
  id: string;
  title: string;
  description: string;
  state: Types.StoryState;
  kind: Types.StoryKind;
  points?: number | undefined;
  requesterId: string;
  projectId: string;
  releaseDate?: DateString | undefined;
  position: Types.StoryPosition;
  priority: number;
  createdAt: DateString;
  updatedAt: DateString;
  isUnEstimated: boolean;
  isCompleted: boolean;
  isDeleted?: boolean | undefined;
  canEstimate: boolean;
};

export type StoryCreateForm_CreateStoryMutationVariables = Types.Exact<{
  input: Types.CreateStoryInput;
}>;

export type StoryCreateForm_CreateStoryMutation = {
  __typename?: 'Mutation';
  createStory:
    | {
        __typename?: 'CreateStorySuccessResult';
        result: {
          __typename?: 'Story';
          id: string;
          title: string;
          description: string;
          state: Types.StoryState;
          kind: Types.StoryKind;
          points?: number | undefined;
          requesterId: string;
          projectId: string;
          releaseDate?: DateString | undefined;
          position: Types.StoryPosition;
          priority: number;
          createdAt: DateString;
          updatedAt: DateString;
          isUnEstimated: boolean;
          isCompleted: boolean;
          isDeleted?: boolean | undefined;
          canEstimate: boolean;
        };
        effectedStories: Array<{
          __typename?: 'Story';
          id: string;
          priority: number;
        }>;
      }
    | { __typename?: 'InternalErrorResult' }
    | {
        __typename?: 'InvalidArgumentsResult';
        issues: Array<{
          __typename?: 'ValidationIssue';
          field?: string | undefined;
          message?: string | undefined;
        }>;
      }
    | { __typename?: 'UnauthorizedResult' };
};

export const StoryCreateForm_ItemFragmentDoc = gql`
  fragment StoryCreateForm_Item on Story {
    id
    title
    description
    state
    kind
    points
    requesterId
    projectId
    releaseDate
    position
    priority
    createdAt
    updatedAt
    isUnEstimated
    isCompleted
    isDeleted
    canEstimate
  }
`;
export const StoryCreateForm_CreateStoryDocument = gql`
  mutation StoryCreateForm_CreateStory($input: CreateStoryInput!) {
    createStory(input: $input) {
      ... on CreateStorySuccessResult {
        result {
          ...StoryCreateForm_Item
        }
        effectedStories {
          id
          priority
        }
      }
      ... on InvalidArgumentsResult {
        issues {
          field
          message
        }
      }
    }
  }
  ${StoryCreateForm_ItemFragmentDoc}
`;

export function useStoryCreateForm_CreateStoryMutation() {
  return Urql.useMutation<
    StoryCreateForm_CreateStoryMutation,
    StoryCreateForm_CreateStoryMutationVariables
  >(StoryCreateForm_CreateStoryDocument);
}
