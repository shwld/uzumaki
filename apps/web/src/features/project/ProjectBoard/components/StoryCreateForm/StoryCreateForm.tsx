import { Box, VStack } from '@chakra-ui/react';
import { zodResolver } from '@hookform/resolvers/zod';
import { generateId } from 'core-domain';
import { StoryPosition } from 'graphql-resolvers/src/generated/resolvers-types';
import { FC } from 'react';
import { useForm } from 'react-hook-form';
import {
  InputMaybe,
  Scalars,
  StoryKind,
  StoryState,
} from '~/graphql/generated/graphql';
import { valueAsDate } from '~/shared/functions/valueAsDate';
import { valueAsNumber } from '~/shared/functions/valueAsNumber';
import { valueAsString } from '~/shared/functions/valueAsString';
import { StoryButtons } from '../StoryForm/StoryButtons';
import { StoryDescriptionInput } from '../StoryForm/StoryDescriptionInput';
import { StoryKindInput } from '../StoryForm/StoryKindInput';
import { StoryPointsInput } from '../StoryForm/StoryPointsInput';
import { StoryReleaseDateInput } from '../StoryForm/StoryReleaseDateInput';
import { StoryRequesterInput } from '../StoryForm/StoryRequesterInput';
import { StoryStateInput } from '../StoryForm/StoryStateInput';
import { StoryTitleInput } from '../StoryForm/StoryTitleInput';
import { useStoryCreateForm_CreateStoryMutation } from './StoryCreateForm.generated';
import { createStoryArgsValidationSchema } from 'graphql-resolvers/src/modules/story/mutation-resolvers/story.create/create-story-validation';

type StoryInput = {
  id: Scalars['ID'];
  title: Scalars['String'];
  state: StoryState;
  description: Scalars['String'];
  kind: StoryKind;
  points?: InputMaybe<Scalars['Int']>;
  releaseDate?: InputMaybe<Scalars['DateTime']>;
  requesterId: Scalars['ID'];
  position: StoryPosition;
  priority: Scalars['Int'];
  projectId: Scalars['ID'];
};

export const StoryCreateForm: FC<{
  projectId: string;
  destination: {
    position: StoryPosition;
    priority: number;
  };
  onCancel?(): void;
  onComplete?(): void;
}> = ({ projectId, destination, onCancel, onComplete }) => {
  const [createStoryResult, createStory] =
    useStoryCreateForm_CreateStoryMutation();
  const {
    handleSubmit,
    register,
    formState: { errors },
  } = useForm<StoryInput>({
    resolver: zodResolver(createStoryArgsValidationSchema.shape.input),
    defaultValues: {
      title: '',
      state: StoryState.Unstarted,
      points: undefined,
      position: destination.position,
      priority: destination.priority,
      id: generateId(),
      projectId,
    },
  });
  const submit = handleSubmit(async input => {
    await createStory({
      input,
    });
    onComplete && onComplete();
  });
  return (
    <Box p={3} bg="orange.100">
      <form onSubmit={submit}>
        <input type="hidden" {...register('id')} />
        <input type="hidden" {...register('projectId')} />
        <input type="hidden" {...register('position')} />
        <input type="hidden" {...register('priority')} />
        <StoryTitleInput {...register('title')} errors={errors} />

        <StoryButtons
          disabled={createStoryResult.fetching}
          onCancel={onCancel}
        />

        <StoryStateInput errors={errors} {...register('state')} />

        <VStack align="stretch" rounded="md" bg="white" mt={3} py={1} gap={2}>
          <StoryKindInput {...register('kind')} errors={errors} />

          <StoryReleaseDateInput
            {...register('releaseDate', { setValueAs: valueAsDate })}
            errors={errors}
          />

          <StoryPointsInput
            {...register('points', {
              setValueAs: valueAsNumber,
            })}
            errors={errors}
          />

          <StoryRequesterInput
            projectId={projectId}
            {...register('requesterId', { setValueAs: valueAsString })}
            errors={errors}
          />
        </VStack>

        <StoryDescriptionInput {...register('description')} errors={errors} />
      </form>
    </Box>
  );
};
