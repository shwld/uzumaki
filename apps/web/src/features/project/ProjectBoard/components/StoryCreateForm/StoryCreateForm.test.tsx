import { StoryCreateForm } from './StoryCreateForm';
import { render } from '@testing-library/react';
import { MockedUrqlProvider } from '~/test/MockedUrqlProvider';
import { StoryPosition } from 'graphql-resolvers/src/generated/resolvers-types';

describe('StoryCreateForm', () => {
  const renderComponent = () => {
    const renderResult = render(
      <MockedUrqlProvider>
        <StoryCreateForm
          projectId="test"
          destination={{ position: StoryPosition.Icebox, priority: 0 }}
        />
      </MockedUrqlProvider>
    );
    return renderResult;
  };
  test('Snapshot', () => {
    expect(renderComponent().asFragment()).toMatchSnapshot();
  });
  test('success', () => {
    const { getByText } = renderComponent();
    expect(getByText('Save')).toBeTruthy();
  });
});
