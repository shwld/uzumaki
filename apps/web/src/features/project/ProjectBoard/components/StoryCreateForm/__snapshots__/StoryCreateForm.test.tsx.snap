// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryCreateForm Snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1w92uxk"
  >
    <form>
      <div
        class="chakra-stack css-1h4ws66"
      >
        <div
          class="chakra-stack css-1nwgsmj"
        >
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <input
              class="chakra-input css-cxmp86"
              id="field-1"
              name="title"
              type="text"
            />
          </div>
        </div>
      </div>
      <div
        class="chakra-stack css-rcjhfu"
      >
        <div
          class="chakra-button__group css-1ayoy1w"
          role="group"
        />
        <div
          class="chakra-stack css-2nt6bn"
        >
          <button
            class="chakra-button css-5c2176"
            type="submit"
          >
            Save
          </button>
        </div>
      </div>
      <div
        class="chakra-stack css-ysmzeb"
      >
        <div
          class="chakra-stack css-yemcr4"
        >
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <div
              class="css-70qvj9"
            >
              <label
                class="chakra-form__label css-1w9kk21"
                for="state"
                id="field-2-label"
              >
                State
              </label>
              <div
                class="chakra-select__wrapper css-42b2qy"
              >
                <select
                  class="chakra-select css-1ik61og"
                  id="state"
                  name="state"
                >
                  <option
                    value="UNSTARTED"
                  >
                    UNSTARTED
                  </option>
                  <option
                    value="STARTED"
                  >
                    STARTED
                  </option>
                  <option
                    value="FINISHED"
                  >
                    FINISHED
                  </option>
                  <option
                    value="ACCEPTED"
                  >
                    ACCEPTED
                  </option>
                  <option
                    value="REJECTED"
                  >
                    REJECTED
                  </option>
                </select>
                <div
                  class="chakra-select__icon-wrapper css-162mkon"
                >
                  <svg
                    aria-hidden="true"
                    class="chakra-select__icon"
                    focusable="false"
                    role="presentation"
                    style="width: 1em; height: 1em; color: currentColor;"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="chakra-stack css-ysmzeb"
      >
        <div
          class="chakra-stack css-yemcr4"
        >
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <div
              class="css-70qvj9"
            >
              <label
                class="chakra-form__label css-1w9kk21"
                for="kind"
                id="field-3-label"
              >
                TYPE
              </label>
              <div
                class="chakra-select__wrapper css-42b2qy"
              >
                <select
                  class="chakra-select css-1ik61og"
                  id="kind"
                  name="kind"
                >
                  <option
                    value="FEATURE"
                  >
                    Feature
                  </option>
                  <option
                    value="BUG"
                  >
                    Bug
                  </option>
                  <option
                    value="CHORE"
                  >
                    Chore
                  </option>
                  <option
                    value="RELEASE"
                  >
                    Release
                  </option>
                </select>
                <div
                  class="chakra-select__icon-wrapper css-162mkon"
                >
                  <svg
                    aria-hidden="true"
                    class="chakra-select__icon"
                    focusable="false"
                    role="presentation"
                    style="width: 1em; height: 1em; color: currentColor;"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="chakra-stack css-yemcr4"
        >
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <div
              class="css-70qvj9"
            >
              <label
                class="chakra-form__label css-1w9kk21"
                for="releaseDate"
                id="field-4-label"
              >
                RELEASE DATE
              </label>
              <input
                class="chakra-input css-0"
                id="field-4"
                name="releaseDate"
                type="date"
              />
            </div>
          </div>
        </div>
        <div
          class="chakra-stack css-yemcr4"
        >
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <div
              class="css-70qvj9"
            >
              <label
                class="chakra-form__label css-1w9kk21"
                for="points"
                id="field-5-label"
              >
                POINTS
              </label>
              <div
                class="chakra-select__wrapper css-42b2qy"
              >
                <select
                  class="chakra-select css-1ik61og"
                  id="field-5"
                  name="points"
                >
                  <option>
                    Unestimated
                  </option>
                  <option
                    value="1"
                  >
                    1
                  </option>
                  <option
                    value="2"
                  >
                    2
                  </option>
                  <option
                    value="3"
                  >
                    3
                  </option>
                  <option
                    value="5"
                  >
                    5
                  </option>
                  <option
                    value="8"
                  >
                    8
                  </option>
                  <option
                    value="13"
                  >
                    13
                  </option>
                  <option
                    value="20"
                  >
                    20
                  </option>
                  <option
                    value="40"
                  >
                    40
                  </option>
                </select>
                <div
                  class="chakra-select__icon-wrapper css-162mkon"
                >
                  <svg
                    aria-hidden="true"
                    class="chakra-select__icon"
                    focusable="false"
                    role="presentation"
                    style="width: 1em; height: 1em; color: currentColor;"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="chakra-stack css-yemcr4"
        >
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <div
              class="css-70qvj9"
            >
              <label
                class="chakra-form__label css-1w9kk21"
                for="requesterId"
                id="field-6-label"
              >
                REQUESTER
              </label>
              <div
                class="chakra-select__wrapper css-42b2qy"
              >
                <select
                  class="chakra-select css-1ik61og"
                  id="field-6"
                  name="requesterId"
                >
                  <option
                    value="d9d74e52-e374-4a7a-94e3-ea15097779b8"
                  >
                    et
                  </option>
                </select>
                <div
                  class="chakra-select__icon-wrapper css-162mkon"
                >
                  <svg
                    aria-hidden="true"
                    class="chakra-select__icon"
                    focusable="false"
                    role="presentation"
                    style="width: 1em; height: 1em; color: currentColor;"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="chakra-stack css-zvppbj"
      >
        <div
          class="chakra-form-control css-0"
          role="group"
        >
          <label
            class="chakra-form__label css-uvho5g"
            for="description"
            id="field-7-label"
          >
            DESCRIPTION
          </label>
          <textarea
            class="chakra-textarea css-cxmp86"
            id="field-7"
            name="description"
          />
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
